### ISSUE TITLE
Debug console log source messages cut off text

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#46777]
I've never noticed this before. The text always wraps correctly except for the last line - if it runs into the log source on the right side, then we lose some text.

![image](https://user-images.githubusercontent.com/323878/37998847-27811a4c-31d5-11e8-9f14-36dd6f86e902.png)

**Comment by isidorn** [comment]
Yeah the log sources are screwing up my wrapping strategies. Nothing new, april to investigate how to fix

**Comment by isidorn** [comment]
Now we are taking into account the width of the source link on the right.
Please note that the wrapping strategy is still not perfect, but it behaves good in most cases.

**Comment by weinand** [comment]
Seems to work fine:
![2018-04-25 16 44 15](https://user-images.githubusercontent.com/1898161/39253393-082d2854-48a8-11e8-8ec3-4be175242411.gif)
